TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py

```diff
@@ -86,23 +88,38 @@ def __init__(self, config: Dict) -> None:
         elif 'sigv4' in config:
             self.auth_type = AuthMethod.SIGV4
 
+    def _get_basic_auth_password(self) -> str:
+        assert self.auth_type == AuthMethod.BASIC_AUTH
+        assert self.auth_details is not None  # for mypy's sake
+        if "password" in self.auth_details:
+            return self.auth_details["password"]
+        # Pull password from AWS Secrets Manager
+        assert "password_from_secret_arn" in self.auth_details
+        client = boto3.client('secretsmanager')
+        password = client.get_secret_value(SecretId=self.auth_details["password_from_secret_arn"])
+        return password["SecretString"]
```

Comment:
I don't think this is a blocker for this PR since it's such a big possible change, but I'm curious to get your thoughts here.  It feels odd to mix input validation and actual work the way this module does; typically you validation everything before doing work as separate layers.  Is there a way to split out the validation so it's all accomplished before any work is done?  It feels like that could make the code easier to read and test.

Maybe we end up with two classes - a ClusterConfig type that takes in the schema and validates its contents and a Cluster type that takes in a reference to a ClusterConfig in order to do things like the `call_api` method.  Or something like that?

Reply:
First of all, is this comment on these specific lines of code because of the `assert` line? I should add a comment to it, but the config has already been validated, etc. so there's no risk of not having this value, but mypy (type hint checking) is not equally convinced of that unless I add this line.

Reply:
So I don't think we're actually mixing in any input validation. It is doing all the validation before any work is done.

Validation is the schema + lines ~74-78 above.

Reply:
I'll call out why I'm doing the assert though.

Reply:
Sorry, this is just were I ended up sticking the comment.  It's really about the entire structure of the responsibilities of the class and, I suspect, this layer of the code in general.

Reply:
I agree that the class does both validation and functional work.

But to your point:
>  Is there a way to split out the validation so it's all accomplished before any work is done? It feels like that could make the code easier to read and test.


I think the validation is accomplished before work is done, and I think it's limited to a pretty small area of the class without any complex logic in it. Effectively, all of that logic is contained in the schema, which I believe basically functions as the Config object you're describing.

Reply:
It sounds like we agree this type is performing two responsibilities then - presenting a config spec/validating adherence AND providing operations using that config.  The classic argument of why you'd split these into two types is if you need two very similar but different behaviors for the same config - such as different ways to interact with versioned Elasticsearch/Opensearch clusters.  This seems like an issue we will run into in the near future.

Let's talk about this one more; I don't think we need to resolve the issue in this PR but having some discussion on how we approach the topic going forward could save us time/effort later.

Reply:
Per discussion w/ @mikaylathompson - while I feel like splitting the responsibilities into separate types and using composition to re-incorporate them is the better approach long term due to the isolation and flexibility it provides:
* The current code isn't unreasonable or hard to read
* The tests are currently simple
* There is a path forward to provide different behavior for different versions w/ inheritance

For these reasons, we'll proceed with the current path for the time being until it becomes an issue.